### ISSUE TITLE
What is this notification type, and will it ever recur? [Wrapstodon]

### ISSUE THREAD

**Issue by mcclure** [mastodon/mastodon#33152]
### Steps to reproduce the problem

I was reading Mastodon.social via the Android app Tusky. I saw this

![Image](https://github.com/user-attachments/assets/4b628e6c-e4fc-489d-97c8-0ff22ca1fd5f)

I was surprised. Had I found a bug in Tusky? So I opened mastodon.social in a web browser and I scrolled down to the Greg Parker message:

![Image](https://github.com/user-attachments/assets/a2768fe3-6532-4b92-aa09-8fb977e11c74)

Oh. That thing.

It appears your new "Wrapstodon" thing uses a unique notification type which third-party clients seem to require specific support for. As a member of the Tusky project, I have questions:

- What is the notification type? Is it documented? *Will* it be documented?
- Is this notification type something that will recur again, besides this one-off event on December 2, 2024? If so is it specific to your yearly "wrapstodon" event or or is this actually some kind of general server-staff announcement notification type that might be important for other purposes as well?
 
We have not attempted to perform manual debugging such as extracting the returned notification JSON.

### Expected behaviour

-

### Actual behaviour

-

### Mastodon instance

mastodon.social

### Mastodon version

v4.4.0-nightly.2024-12-02

### Technical details

If this is happening on your own Mastodon server, please fill out those:
- Ruby version: (from `ruby --version`, eg. v3.3.5)
- Node.js version: (from `node --version`, eg. v20.18.0)

**Comment by renchap** [comment]
> What is the notification type? Is it documented? Will it be documented?

It is not documented. The feature is not finished, but it has been decided to enable it on mastodon.social as it wont be fully finished before the end of the year, and this feature is mostly useful in december.

Once finished, it will be documented. At the moment, clients should ignore it (and we recommend that client ignore unknown notification types by default probably?), and the notification does not contain much useful information. The underlying data is available through an API call, so you could build your own representation for it. The feature is intended to have a link to an "embeddable" Wrapstodon page at some point, which client could display in a webview to avoid having to reimplement it.

> Is this notification type something that will recur again, besides this one-off event on December 2, 2024? If so is it specific to your yearly "wrapstodon" event or or is this actually some kind of general server-staff announcement notification type that might be important for other purposes as well?

This notification type is specific to Wrapstodon, and will happen every year when an admin generate the statistics (which currently require some custom console code)

**Comment by mistydemeo** [comment]
With this year's wrapstodon, I noticed that this notification type is still undocumented. The `annual_report` type doesn't appear anywhere in the API documents. Is the API complete? Is there a plan to document this so other clients can display it?

**Comment by renchap** [comment]
Yes, we plan to write API documentation for this feature but we wanted to first get feedback on this iteration to see if it needs adjustments before we make this an official API.

**Comment by jsoref** [comment]
- Afaict, this hasn't triggered for 2025. Was it going to? My initial feedback is #38338

**Comment by renchap** [comment]
This notification type has been deprecated and will no longer be emitted. 

I am closing this issue due to this, the new Wrapstodon implementation does not use notifications anymore.

I am keeping #38338 open so we properly handle this in the webapp.
